### ISSUE TITLE
Unintuitive back button behavior

### ISSUE THREAD

**Issue by agersant** [agersant/polaris-android#40]
Using latest from `master`.

Repro steps:
1. Navigate to a Collection folder
2. Press the Queue button
3. Press the Now Playing button
4. Press the device back button

Expected result:
- User is returned to the Queue screen. Pressing back a second time returns to the Collection folder

Actual result:
- User is returned to the Collection folder

cc @Saecki if you have time/interest to look into this

**Comment by Saecki** [comment]
Yes I'm interested. I already fixed this using Viewmodels to cache data. Once I iron out a few remaining bugs I will open a PR.

**Comment by saecki** [comment]
I think that has to do with the top-level destinations inside the `BottomNavigationView` and how the navigation handles them. The navigation stack will not store transitions between top-level destinations when pressing the icon on the `BottomNavigationView`. Instead we have separate backstacks for every top level destination. Since `Collection` is the start destination of our navigation graph pressing back on another top-level destionation will bring you back to it and not to `Queue` or `Now Playing`.

The fix for this would be to not use multiple backstacks on the top-level destionations but to use one backstack that would track every navigation action. That would also result in following behavior:

1. Navigate to a Collection folder
2. Press the Queue button
3. Press the Collection button
4. Press the back button

The User is returned to the Queue instead of the parent directory.

I personally think that might be more confusing than the current behavior.

Another note:
In android navigation `2.4` the multiple backstack behavior that is currently implemented using the `NavigationExtensions` will become default. The current release `2.4.0-alpha02` however still has some bugs that make it a pain to use. I will open an issue to keep track of that.

**Comment by agersant** [comment]
Outdated issue, UI has changed drastically since then.
